File: Padrões Gof/Estruturais/Proxy/Proxy01.js
// Objeto Real
class ServicoReal{
  executar(){
    console.log("Executando o serviço real...")
  }
}

// Proxy
class ProxyServico{
  constructor(){
    this.servicoReal = new ServicoReal();
  }
  
  executar(){
    console.log("Antes da execução do serviço");
    this.servicoReal.executar();
    console.log("Depois da execução do serviço...")
  }
}


// USO DO PROXY - CLIENTE #####
const proxy = new ProxyServico();
proxy.executar();

File: Padrões Gof/Estruturais/Proxy/Proxy01.py
# Objeto Real
class ServicoReal:
    def executar(self):
        print("Executando o serviço real...")

# Proxy
class ProxyServico:
    def __init__(self):
        self.servicoReal = ServicoReal()

    def executar(self):
        print("Antes da execução do serviço")
        self.servicoReal.executar()
        print("Depois da execução do serviço...")

# USO DO PROXY - CLIENTE
proxy = ProxyServico()
proxy.executar()
